src/content/docs/en/guides/backend/google-firebase.mdx

```diff
@@ -11,6 +11,432 @@ import FileTree from '~/components/FileTree.astro'
+:::
+
+Finally, your project directory should now include these new files:
+
+<FileTree title="Project Structure">
+- src
+  - env.d.ts
+  - firebase
+    - client.ts
+    - server.ts
+  - pages
+    - api
+      - auth
+        - **login.ts**
+        - **logout.ts**
+        - **register.ts**
+- .env
+- astro.config.mjs
+- package.json
+</FileTree>
+
+### Creating pages
+
+Now that we have authentication endpoints set up, we can create the pages that will use them. We will create three new pages:
+
+- `pages/register` - will contain a form to register a user
+- `pages/dashboard` - will contain a dashboard that can only be accessed by authenticated users
+- `pages/signin` - will contain a form to sign in a user
+
+Create a file named `register.astro` inside the `pages` folder.
+
+```astro title="src/pages/register.astro"
+---
+import Layout from "../layouts/Layout.astro";
+---
+<Layout title="Register">
+	<h1>Register</h1>
+  <p>Already have an account? <a href="/signin">Sign in</a></p>
+	<form action="/api/auth/register" method="post">
+    <label for="name">Name</label>
+    <input type="text" name="name" id="name" />
+    <label for="email" for="email">Email</label>
+		<input type="email" name="email" id="email" />
+    <label for="password">Password</label>
+		<input type="password" name="password" id="password" />
+		<button type="submit">Login</button>
+	</form>
+</Layout>
+```
+
+The markup of the page contains a form that will send a `POST` request to the `/api/auth/register` endpoint. The endpoint will create a new user using the data from the form and redirect the user to the `/signin` page.
+
+Next, create a file named `dashboard.astro` inside the `pages` folder.
+
+```astro title="src/pages/dashboard.astro"
+---
+import { app } from "../firebase/server";
+import { getAuth } from "firebase-admin/auth";
+import Layout from "../layouts/Layout.astro";
+
+const auth = getAuth(app);
+
+/* Check current session */
+const sessionCookie = Astro.cookies.get("session").value;
+
+if (!sessionCookie) {
+  return Astro.redirect("/signin");
+}
+const decodedCookie = await auth.verifySessionCookie(sessionCookie);
+const user = await auth.getUser(decodedCookie.uid);
+
+if (!user) {
+  return Astro.redirect("/signin");
+}
+---
+<Layout title="dashboard">
+	<h1>Welcome {user.displayName}</h1>
+	<p>We are happy to see you here</p>
+  <form action="/api/auth/signout">
+    <button type="submit">Sign out</button>
+  </form>
+</Layout>
+</html>
+```
+
+In the frontmatter of the page, using the Firebase server app we verify the user's session cookie. If the user is not authenticated, the page will redirect the user to the `/signin` page.
+
+In the markup of the page, we display the user's name and a button to sign out. The signout button will send a `GET` request to the `/api/auth/signout` endpoint. The endpoint will delete the user's session cookie and redirect the user to the `/signin` page.
```

Comment:
I believe it should be "sign out" with whitespace. Considering the entire sentence, I would simply remove this second mention of the word.
```suggestion
In the markup of the page, we display the user's name and a button to sign out. The button will send a `GET` request to the `/api/auth/signout` endpoint. The endpoint will delete the user's session cookie and redirect the user to the `/signin` page.
```